semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -1736,6 +1738,17 @@ and m_stmts_deep_uncached ~less_is_ok (xsa : G.stmt list) (xsb : G.stmt list) =
   | [], _ :: _ -> if less_is_ok then return () else fail ()
   (* dots: '...', can also match no statement *)
   | [ { s = G.ExprStmt ({ e = G.Ellipsis _i; _ }, _); _ } ], [] -> return ()
+  (* inside:
+   * When a pattern-inside (or pattern-not-inside) ends in ... we do not
+   * need to compute all possible endings, it's enough to return the largest
+   * one. If some other pattern P matches inside the largest match here, then
+   * that is enough to either keep P (pattern-inside) or filter it out
+   * (pattern-not-inside).
+   *)
+  | [ { s = G.ExprStmt ({ e = G.Ellipsis _i; _ }, _); _ } ], xb :: bbs
```

Comment:
Great comment, and optimization. Good to take advantage of knowing that the pattern is for a `pattern-inside`